s-ramp-ide/srampRepositoryBrowser/src/main/java/cz/muni/fi/srampRepositoryBrowser/dialogs/Filter.java

```diff
@@ -0,0 +1,529 @@
+		lastModifiedToT.addMouseListener(new MouseListener() {
+
+			@Override
+			public void mouseDown(MouseEvent arg0) {
+
+				DateTimeDialog dt = new DateTimeDialog(getShell(), SWT.NONE,
+						lastModifiedToT.toDisplay(0, 0));
+				lastModifTo = dt.open();
+				DateFormat dateFormat = new SimpleDateFormat("MM/dd/YYYY");
+				lastModifiedToT.setText(dateFormat.format(lastModifTo.getTime()));
+			}
+
+			@Override
+			public void mouseDoubleClick(MouseEvent e) {
+			}
+
+			@Override
+			public void mouseUp(MouseEvent e) {
+			}
+
+		});
+
+		Label createdBy = new Label(this, SWT.NONE);
+		createdBy.setEnabled(true);
+		createdBy.setText("Created By");
+		new Label(this, SWT.NONE);
+		new Label(this, SWT.NONE);
+
+		createdByT = new Text(this, SWT.BORDER);
+		createdByT.setEnabled(true);
+		GridData gd_createdByT = new GridData(SWT.FILL, SWT.CENTER, false,
+				false, 3, 1);
+		gd_createdByT.widthHint = 230;
+		createdByT.setLayoutData(gd_createdByT);
+
+		Label lastModifiedBy = new Label(this, SWT.NONE);
+		lastModifiedBy.setEnabled(true);
+		lastModifiedBy.setText("Last Modified By");
+		new Label(this, SWT.NONE);
+		new Label(this, SWT.NONE);
+
+		lastModifiedByT = new Text(this, SWT.BORDER);
+		lastModifiedByT.setEnabled(true);
+		GridData gd_lastModifiedByT = new GridData(SWT.FILL, SWT.CENTER, false,
+				false, 3, 1);
+		gd_lastModifiedByT.widthHint = 230;
+		lastModifiedByT.setLayoutData(gd_lastModifiedByT);
+
+		btnFilter = new Button(this, SWT.NONE);
+		btnFilter.setEnabled(true);
+		btnFilter.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER, false,
+				false, 1, 1));
+		btnFilter.setText("Filter");
+		btnFilter.addSelectionListener(new SelectionAdapter() {
+			public void widgetSelected(SelectionEvent event) {
+				Filter.this.parent.setFilter(setFilter());
+				Job refresh = new RefreshJob("refreshing", Filter.this.parent);
+				refresh.schedule();
+			}
+		});
+		new Label(this, SWT.NONE);
+
+		btnClearAllFilters = new Button(this, SWT.NONE);
+		btnClearAllFilters.setEnabled(true);
+		btnClearAllFilters.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER,
+				false, false, 1, 1));
+		btnClearAllFilters.setText("Clear all filters");
+		btnClearAllFilters.addSelectionListener(new SelectionAdapter() {
+			public void widgetSelected(SelectionEvent event) {
+				Filter.this.parent.setFilter(Filter.this.parent.getManager()
+						.listAllArtifacts());
+				typeT.setText("");
+				createdByT.setText("");
+				lastModifiedByT.setText("");
+				createdFrom = null;
+				createdTo = null;
+				lastModifFrom = null;
+				lastModifTo = null;
+
+				createdFromT.setText("Any");
+				createdToT.setText("Any");
+				lastModifiedFrom.setText("Any");
+				lastModifiedToT.setText("Any");
+
+				Job refresh = new RefreshJob("refreshing", Filter.this.parent);
+				refresh.schedule();
+			}
+		});
+
+	}
+
+	/**
+	 * Find artifact model from artifact type.
+	 * 
+	 * @param type
+	 * @return artifact model
+	 */
+	private String findModel(String type) {
```

Comment:
Take a look at `ArtifactType#valueOf(String)` in `s-ramp-common`.  Pass that `type`, then you'll be able to do `artifactType#getModel`